app/assets/javascripts/models/beta/projectBoard.js

```diff
@@ -68,44 +68,72 @@ export const toggleColumn = (projectBoard, column, callback) => {
 }
 
 // Drag And drop utils
+const calculateNewPosition = (aboveStory, bellowStory, storyState) => {
```

Comment:
@petruspierre This calculateNewPosition and calculatePosition are very similar functions, we have 4 possible different outcomes and 10 different returns, maybe this function could be given more thought into it, although I dont have a suggestion right now on how to improve it, it you mind explaining this logic to me I can help you with that :+1: